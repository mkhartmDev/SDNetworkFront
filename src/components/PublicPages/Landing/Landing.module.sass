@import '../../../index'

.LandingPage
    display: flex
    flex-direction: column
    justify-content: center
    align-items: center
    gap: 30px
    height: 100vh
    background-color: $faded-base-color

.Controls
    background-color: $subtle-offwhite-coloring
    border: 2px solid black
    padding: 50px
    border-radius: 4px
    box-shadow: 0px 8px 16px 0px #0000005e
    margin-bottom: 300px

.WelcomeMessage
    padding-bottom: 20px
    margin-bottom: 20px
    font-size: 22px

.ButtonsContainer
    display: flex
    flex-direction: row
    justify-content: center
    align-items: center
    gap: 45px

.NavLink
    text-decoration: none
    color: white
    background-color: $base-color
    border: 2px solid white
    border-radius: 4px
    width: 200px
    height: 30px
    display: flex
    flex-direction: column
    justify-content: center
    align-items: center

.NavLink:hover
    background-color: $hover-color

.NavLink:active
    background-color: $active-color