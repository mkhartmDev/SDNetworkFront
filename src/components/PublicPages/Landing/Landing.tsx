import React, { Component } from 'react'
import classes from './Landing.module.sass'
import { NavLink } from 'react-router-dom'

interface Props {
    
}
interface State {
    
}

export default class Landing extends Component<Props, State> {
    state = {}

    render() {
        return (
            <div className={classes.LandingPage}>
                <div className={classes.Controls}>
                    <div className={classes.WelcomeMessage}>Welcome to the Social Distancing Network</div>
                    <div className={classes.ButtonsContainer}>
                        <NavLink className={classes.NavLink} to='/create-an-account'>Create An Account</NavLink>
                        <NavLink className={classes.NavLink} to='/log-in'>Log In</NavLink>
                    </div>
                </div>
            </div>
        )
    }
    
}
