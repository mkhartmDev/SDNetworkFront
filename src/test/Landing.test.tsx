import React from 'react';
import {shallow,mount,render} from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Landing from '../components/PublicPages/Landing/Landing';
import {MemoryRouter, Router}  from 'react-router-dom';




Enzyme.configure({adapter:new Adapter()})



test('render the landing page', () => {
  //Component surrounded by router so it didn't complain about the navlinks inside
  const component =render(<MemoryRouter initialEntries={[ '/' ]}>
    <Landing/>
    </MemoryRouter>)
    expect(component).toMatchSnapshot();
});