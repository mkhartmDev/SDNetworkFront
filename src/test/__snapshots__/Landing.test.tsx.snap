// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounts the component 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "LandingPage",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Controls",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "WelcomeMessage",
            },
            "children": Array [
              Object {
                "data": "Welcome to the Social Distancing Network",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "ButtonsContainer",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/log-in",
                    },
                    "children": Array [
                      Object {
                        "data": "Log In",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/create-an-account",
                    },
                    "children": Array [
                      Object {
                        "data": "Create An Account",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "ButtonsContainer",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/create-an-account",
                },
                "children": Array [
                  Object {
                    "data": "Create An Account",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/log-in",
                },
                "children": Array [
                  Object {
                    "data": "Log In",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "WelcomeMessage",
              },
              "children": Array [
                Object {
                  "data": "Welcome to the Social Distancing Network",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`render the component 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "LandingPage",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Controls",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "WelcomeMessage",
            },
            "children": Array [
              Object {
                "data": "Welcome to the Social Distancing Network",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "ButtonsContainer",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/log-in",
                    },
                    "children": Array [
                      Object {
                        "data": "Log In",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/create-an-account",
                    },
                    "children": Array [
                      Object {
                        "data": "Create An Account",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "ButtonsContainer",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/create-an-account",
                },
                "children": Array [
                  Object {
                    "data": "Create An Account",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/log-in",
                },
                "children": Array [
                  Object {
                    "data": "Log In",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "WelcomeMessage",
              },
              "children": Array [
                Object {
                  "data": "Welcome to the Social Distancing Network",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`render the landing page 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "LandingPage",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Controls",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "WelcomeMessage",
            },
            "children": Array [
              Object {
                "data": "Welcome to the Social Distancing Network",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "ButtonsContainer",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/log-in",
                    },
                    "children": Array [
                      Object {
                        "data": "Log In",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "NavLink",
                      "href": "/create-an-account",
                    },
                    "children": Array [
                      Object {
                        "data": "Create An Account",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "ButtonsContainer",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/create-an-account",
                },
                "children": Array [
                  Object {
                    "data": "Create An Account",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/log-in",
                  },
                  "children": Array [
                    Object {
                      "data": "Log In",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "NavLink",
                  "href": "/log-in",
                },
                "children": Array [
                  Object {
                    "data": "Log In",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "NavLink",
                    "href": "/create-an-account",
                  },
                  "children": Array [
                    Object {
                      "data": "Create An Account",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "a",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "href": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "WelcomeMessage",
              },
              "children": Array [
                Object {
                  "data": "Welcome to the Social Distancing Network",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`renders the component 1`] = `ShallowWrapper {}`;
